src/vis/bar/BarVis.tsx

```diff
@@ -0,0 +1,212 @@
+        return merge({}, defaultExtensions, extensions);
+    }, []);
+
+    const traces: PlotlyInfo = useMemo(() => {
+        return createBarTraces(columns, config, scales);
+    }, [columns, config, scales]);
+
+    const layout = useMemo(() => {
+        const layout = {
+            showlegend: true,
+            legend: {
+                itemclick: false,
+                itemdoubleclick: false
+            },
+            autosize: true,
+            grid: {rows: traces.rows, columns: traces.cols, xgap: .3, pattern: 'independent'},
+            shapes: [],
+            violingap: 0,
+            barmode: config.groupType === EBarGroupingType.STACK ? 'stack' : 'group'
+        };
+
+        return beautifyLayout(traces, layout);
+    }, [traces, config.groupType]);
+
+    return (
+        <div className="d-flex flex-row w-100 h-100">
+            <div className="position-relative d-flex justify-content-center align-items-center flex-grow-1">
+                {mergedExtensions.prePlot}
+                {traces.plots.length > 0 ?
+                    (<Plot
+                        divId={'plotlyDiv'}
+                        data={[...traces.plots.map((p) => p.data), ...traces.legendPlots.map((p) => p.data)]}
+                        layout={layout as any}
+                        config={{responsive: true, displayModeBar: false}}
+                        useResizeHandler={true}
+                        style={{width: '100%', height: '100%'}}
+                        //plotly redraws everything on updates, so you need to reappend title and
+                        // change opacity on update, instead of just in a use effect
+                        onUpdate={() => {
+                            for(const p of traces.plots) {
+                                d3.select(`g .${(p.data as any).xaxis}title`)
+                                    .style('pointer-events', 'all')
+                                    .append('title')
+                                    .text(p.xLabel);
+
+                                d3.select(`g .${(p.data as any).yaxis}title`)
+                                    .style('pointer-events', 'all')
+                                    .append('title')
+                                    .text(p.yLabel);
+                            }
+                        }}
+                    />) : (<InvalidCols
+                        message={traces.errorMessage} />)
+                }
+                {mergedExtensions.postPlot}
+
+            </div>
+            <div className="position-relative h-100 flex-shrink-1 bg-light">
+                <button className="btn btn-primary-outline" type="button" data-bs-toggle="collapse" data-bs-target="#generalVisBurgerMenu" aria-expanded="true" aria-controls="generalVisBurgerMenu">
+                    <i className="fas fa-bars"/>
+                </button>
+                <div className="collapse show collapse-horizontal" id="generalVisBurgerMenu">
```

Comment:
This `id` attribute must be unique as well. Please udpate the other occurences accordingly.

Reply:
done